src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -174,6 +174,16 @@ def get(self, uid):
         result = self.proxy('GET', 'user', {'uid': uid})
         return self._append_uid(result)
 
+    @Endpoint()
+    @ReadPermission
+    def get_emails(self):
+        emails = []
+        for uid in json.loads(self.list()):
```

Comment:
If you put the contents of the `list()` method into a helper method like `_list()`, you could use `_list()` to get the result without having to load the JSON, as it would return a Python native data structure. This is due to all methods of a `RestController` (list, get, create, ...) are modified to return JSON. That way you could bypass that.